homeassistant/components/tessie/coordinator.py

```diff
@@ -0,0 +1,82 @@
+"""Tessie Data Coordinator."""
+from datetime import timedelta
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp import ClientResponseError, ClientSession
+from tessie_api import get_state
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import TessieStatus
+
+# This matches the update interval Tessie performs server side
+TESSIE_SYNC_INTERVAL = 10
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TessieDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the Tessie API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api_key: str,
+        vin: str,
+        data: dict[str, Any],
+    ) -> None:
+        """Initialize Tessie Data Update Coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Tessie",
+            update_method=self.async_update_data,
+            update_interval=timedelta(seconds=TESSIE_SYNC_INTERVAL),
+        )
+        self.api_key: str = api_key
+        self.vin: str = vin
+        self.session: ClientSession = async_get_clientsession(hass)
```

Comment:
```suggestion
        self.session = async_get_clientsession(hass)
```

typing here is redudant